lib/routes/account.js

```diff
@@ -1026,37 +1073,88 @@ module.exports = function (
         }
       },
       handler: function (request, reply) {
-        log.begin('Account.RecoveryEmailVerify', request)
         var uid = request.payload.uid
         var code = Buffer(request.payload.code, 'hex')
+        var tokenData = {uid: uid}
+
+        log.begin('Account.RecoveryEmailVerify', request)
         db.account(Buffer(uid, 'hex'))
           .then(
             function (account) {
-              // If the account is already verified, they may be e.g.
-              // clicking a stale link.  Silently succeed.
-              if (account.emailVerified) {
-                log.increment('account.already_verified')
-                return true
-              }
```

Comment:
I don't recall what we have this "silently succeed on stale links" behaviour, but we should be cautious about refactoring it away.  Let's see if we can get metrics on this `account.already_verified` event to see how many users might be affected by a behaviour change here.